### PR TITLE
Fix the StoreController and add missing data

### LINKED ISSUE #9577: [BOOM-6046] Controller problem after moving stores fields in store_lang
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-6046).

- _**Reporter:**_ videotape
- _**Created at:**_ Fri, 20 Jul 2018 13:04:07 +0200

<p>In /controllers/front/StoresController the method getStoresForXml didn't get store fields that have been moved to store_lang table (such as store name).</p>

<p>So now when we use ajax method to get stores, there's no store names.</p>
        
* How to reproduce the issue ?

Request <a href="http://YOUR_URL/index.php&controller=stores?ajax=1&all=1">http://YOUR_URL/index.php&amp;controller=stores?ajax=1&amp;all=1</a>

---
#### Describe the bug
In /controllers/front/StoresController the method getStoresForXml didn't get store fields that have been moved to store_lang table (such as store name). 

**PS:** _this link use to work with PS1.6.1.20 (see the second attached screenshot below)_

#### Expected behavior
Triggering this link yourStore/index.php?controller=stores&ajax=1&latitude=25.88674000&longitude=-80.16329200 should display all the store informations in XML format

#### Steps to reproduce

1. **BO**>**CONFIGURE**>**Shop Parameters**>**Contact**
2. Go to **Stores** tab 
3. Click on **edit** button of any store
4. Copy the **Latitude / Longitude**
5. In your Navigator, open a new window and trigger this link with your **Latitude / Longitude**

**`http://YOUR_URL/index.php?controller=stores&ajax=1&latitude=25.88674000&longitude=-80.16329200`**

#### Screenshots

image

#### Additional information

- PrestaShop version: 1.7.7.0
- PHP version: 7.2

---
### PS1.6.1.20:

image

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | ajax for the StoreController is now working properly
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #9577
| How to test?      | Details in the bug report
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.
| Sponsor company | impSolutions

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25834)

### CODE REVIEW

**Review by matks — DISMISSED** [review]

File: controllers/front/StoresController.php [hunk 1]
```diff
@@ -36,7 +36,6 @@ public function init()
     {
         parent::init();
 
-        // StarterTheme: Remove check when google maps v3 is done
         if (!extension_loaded('Dom')) {
```
Comment (Progi1984) [inline controllers/front/StoresController.php]: Remove these lines : the `dom` extension is required for PrestaShop.

Comment (PierreRambaud) [inline controllers/front/StoresController.php]: The whole block can be deleted, it's require in composer.json

File: controllers/front/StoresController.php [hunk 2]
```diff
@@ -126,21 +110,20 @@ public function getStoresForXml()
...
             $address = $this->processStoreAddress($store);
 
-            //$other .= $this->renderStoreWorkingHours($store);
             $newnode->addAttribute('addressNoHtml', strip_tags(str_replace('<br />', ' ', $address)));
             $newnode->addAttribute('address', $address);
-            $newnode->addAttribute('other', $other);
+            $newnode->addAttribute('hours', trim($store['hours']));
             $newnode->addAttribute('phone', $store['phone']);
+            $newnode->addAttribute('fax', $store['fax']);
+            $newnode->addAttribute('note', $store['note']);
```
Comment (matks) [inline controllers/front/StoresController.php:126]: So you handle hours, fax, note, but remove the processing of `$other` ?

**Review by matks — APPROVED** [review]
Code seems valid, have 1 question though.

File: controllers/front/StoresController.php (hunk 2 above)
Comment (kpodemski) [reply controllers/front/StoresController.php:126]: `$other` was never used

Comment (matks) [reply controllers/front/StoresController.php:126]: OK then 👍

File: controllers/front/StoresController.php (hunk 1 above)
Comment (atomiix) [reply controllers/front/StoresController.php]: @PierreRambaud But this allows to show an error message during runtime while requiring the `dom` extension in `composer.json` only throws an error when executing `composer install` 🤔

Comment (kpodemski) [reply controllers/front/StoresController.php]: @atomiix doesn't matter, we use `SimpleXMLElement` not a `DOMDocument` and PS installer is checking for it 👍

**Review by atomiix — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by florine2623** [comment]
Hello @kpodemski ,

Thanks for the PR ! Looks good to me :)

**Comment by Progi1984** [comment]
Thanks @kpodemski & @florine2623

**Pull request merged**